exchange/exchange.go

```diff
@@ -57,15 +74,35 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 	}
 	// Randomize the list of adapters to make the auction more fair
 	randomizeList(liveAdapters)
+	// Process the request to check for targeting parameters.
+	targData := &targetData{
+		targetFlag:	false,
+		lengthMax:	0,
+		priceGranularity:	openrtb_ext.PriceGranularityMedium,
+		winningBids:		make(map[string]*openrtb.Bid, len(bidRequest.Imp)),
+		winningBidders:		make(map[string]openrtb_ext.BidderName, len(bidRequest.Imp)),
+	}
+	requestExt := new(openrtb_ext.ExtRequest)
+	err := json.Unmarshal(bidRequest.Ext, requestExt)
```

Comment:
I just pulled the branch to test w the sample request below. It's throwing an error here for the following sample request because the bidRequest `ext` object isn't sent. Works when I add an empty `ext` object to the request. Is this expected? 

{
  "id": "some-request-id",
  "imp": [
    {
      "id": "my-imp-id",
      "banner": {
    	"format": [
    	  {
    	    "w": 300,
    	    "h": 250
    	  },
    	  {
    	    "w": 300,
    	    "h": 600
    	  }
    	]
      },
      "ext": {
        "appnexus": {
          "placementId": 11327263
        }
      }
    }
  ],
  "test": 1,
  "tmax": 500
}

Reply:
Not expected. I fixed this, so it should work smoothly now.